src/Chainweb/Pact/InternalTypes.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE StrictData #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- |
+-- Module: Chainweb.Pact.InternalTypes
+-- Copyright: Copyright © 2019 Kadena LLC.
+-- License: See LICENSE file
+-- Maintainer: Mark Nichols <mark@kadena.io>
+-- Stability: experimental
+--
+-- Pact Types module for Chainweb
+module Chainweb.Pact.InternalTypes
+  ( MemPoolAccess(..)
+  , PactServiceEnv(..)
+  , PactServiceState(..)
+    -- * types
+  , PactServiceM
+    -- * optics
+  , noopMemPoolAccess
+  , psMempoolAccess
+  , psCheckpointEnv
+  , psSpvSupport
+  , psPublicData
+  , psStateValidated
+  , psPdb
+  , psBlockHeaderDb
+    -- * module exports
+  , module Chainweb.Pact.Backend.Types
+  ) where
+
+import Control.Lens hiding ((.=))
+import Control.Monad.Reader
+import Control.Monad.State.Strict
+
+import Data.Vector (Vector)
+import qualified Data.Vector as V
+
+-- internal pact modules
+
+import Pact.Types.ChainMeta (PublicData(..))
+import Pact.Types.SPV
+
+-- internal chainweb modules
+
+import Chainweb.BlockHash
+import Chainweb.BlockHeader
+import Chainweb.BlockHeaderDB.Types
+import Chainweb.Pact.Backend.Types
+import Chainweb.Payload.PayloadStore.Types
+import Chainweb.Transaction
+
+data PactServiceEnv cas = PactServiceEnv
+  { _psMempoolAccess :: !(Maybe MemPoolAccess)
+  , _psCheckpointEnv :: !CheckpointEnv
+  , _psSpvSupport :: !SPVSupport
+  , _psPublicData :: !PublicData
+  , _psPdb :: PayloadDb cas
```

Comment:
In other places we expose it to be able to switch between persisted db backends for production and in memory for testing.

Since we switched to RocksDB, we only use rocks db in production and testing. But we were planning to re-add an in-memory implementation of the storage API for testing. It just not high up on the priority list.